### PR TITLE
Controller authentication

### PR DESCRIPTION
**Pull request by HasaniH**
Besides pushing the services file, everything here was smooth. Tablet controller has authentication

### CODE REVIEW

File: android/cloudspincontroller/app/google-services.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (mimming) [inline android/cloudspincontroller/app/google-services.json]: avoid checking in this file.  It does not contain any secrets, but it is project specific.

File: android/cloudspincontroller/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/AndroidManifest.xml]: Add a license header to this file in an XML comment

File: android/cloudspincontroller/app/src/main/java/com/google/cloudspincontroller/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package com.google.cloudspincontroller;
+
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/java/com/google/cloudspincontroller/MainActivity.java]: Add a license header to this project

File: android/cloudspincontroller/app/src/main/java/com/google/cloudspincontroller/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == RC_SIGN_IN) {
+            IdpResponse response = IdpResponse.fromResultIntent(data);
+
+            // Successfully signed in
+            if (resultCode == ResultCodes.OK) {
+                //Show message when
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/java/com/google/cloudspincontroller/MainActivity.java]: Add a leading space to comments, and complete that sentence :)

File: android/cloudspincontroller/app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/layout/activity_main.xml]: Add a license header

File: android/cloudspincontroller/app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.google.cloudspincontroller.MainActivity">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Hello World!"
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/layout/activity_main.xml]: Don't forget to replace this on a future PR .  Consider adding a TODO

File: android/cloudspincontroller/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">cloud spin controller</string>
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/values/strings.xml]: This should probably be sentence case

Cloud Spin Controller

File: android/cloudspincontroller/app/src/main/res/values/colors.xml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/values/colors.xml]: Add a license header

File: android/cloudspincontroller/app/src/main/res/values/strings.xml (hunk 7 above)
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/values/strings.xml]: Add a license header

File: android/cloudspincontroller/app/src/main/res/values/styles.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<resources>
+
```
Comment (mimming) [inline android/cloudspincontroller/app/src/main/res/values/styles.xml]: Add a license header

File: android/cloudspincontroller/gradle/wrapper/gradle-wrapper.properties [hunk 10]
```diff
@@ -0,0 +1,6 @@
+#Mon Jun 05 14:09:24 PDT 2017
+distributionBase=GRADLE_USER_HOME
```
Comment (mimming) [inline android/cloudspincontroller/gradle/wrapper/gradle-wrapper.properties]: Add a license header

File: android/cloudspincontroller/gradlew [hunk 11]
```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+
```
Comment (mimming) [inline android/cloudspincontroller/gradlew]: Add a license header

File: android/cloudspincontroller/gradlew.bat [hunk 12]
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
+@rem ##########################################################################
```
Comment (mimming) [inline android/cloudspincontroller/gradlew.bat]: Add a license header

File: android/cloudspincontroller/app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (HasaniH) [inline android/cloudspincontroller/app/src/main/res/layout/activity_main.xml]: Yup! I've added the spinner here on my next changelist

**Pull request merged**